docroot/modules/custom/foia_file/drush/foia_file.drush.inc

```diff
@@ -0,0 +1,174 @@
+  $webformDir = "{$fileDir[0]}/webform/";
+
+  // Temporarily delete similar paths of the webform dir during processing.
+  $trimmedFiles = str_replace($webformDir, '', $scanOutput);
+
+  // Split string into lines. @todo change to preg split \R?
+  $scans = explode("\n", $trimmedFiles);
+
+  foreach ($scans as $scan) {
+
+    // Split the scan line into filename and status.
+    preg_match('/[^ ]*$/', $scan, $wordsArray);
+    // Set status to the last word of the line.
+    $status = $wordsArray[0];
+
+    $expectedStatuses = ['OK', 'Removed.', 'Empty file'];
+    if (!in_array($status, $expectedStatuses)) {
+      \Drupal::logger('foia_file')->warning("An unexpected status was detected in the virus scan output.");
+    }
+
+    // Get relative file path.
+    $scanLength = strlen($scan);
+    $statusLength = strlen($status);
+    // Length of the file name is the remainder of the scan minus the colon and
+    // space characters.
+    $fileNameLength = $scanLength - $statusLength - 2;
+    $fileName = substr($scan, 0, $fileNameLength);
+
+    // Validate path, filename, and extension.
+    $pathParts = pathinfo($fileName);
+    foreach ($pathParts as $pathPart) {
+      if (empty($pathPart)) {
+        $status = 'invalid';
+      }
+    }
+
+    // Log any discrepancies between filesystem and entities.
+    if ($status === 'OK' && !is_file($fileName)) {
+      \Drupal::logger('foia_file')->warning(
+        "The file @fileName was not detected on the filesystem and was marked as clean by the virus scanner.",
+        ['@fileName' => $fileName]
+        );
+    }
+    if ($status === 'Removed.' && is_file($fileName)) {
+      \Drupal::logger('foia_file')->warning(
+        "The file @fileName was detected on the filesystem and a virus was detected in it. The file should have been deleted by the scanner.",
+        ['@fileName' => $fileName]
+      );
+    }
+
+    // Determine file entity id based upon file path & name.
+    $query = \Drupal::entityQuery('file')
+      ->condition('uri', "private://webform/{$fileName}");
+    $fids = $query->execute();
+    $fid = array_values($fids)[0];
+
+    // @todo Validate that only one item in fids array.
+
+    // @todo Nice to have >> verify that submission id is same as file path.
+
+    // Update file entity based upon status.
+    if ($fid) {
+      $fileEntity = FileEntity::load($fid);
+    }
+    else {
+      $fileEntity = NULL;
+    }
+    $doNotScanStatuses = ['clean', 'virus', 'invalid'];
+    if ($fileEntity && !in_array($fileEntity->get('field_virus_scan_status')->getString(), $doNotScanStatuses)) {
+      if ($status === "OK") {
+        $fileEntity->set('field_virus_scan_status', 'clean');
+        \Drupal::logger('foia_file')->info(
+          "File @webformDir@fileName Entity ID @fid set to clean.",
+          [
+            '@webformDir' => $webformDir,
+            '@fileName' => $fileName,
+            '@fid' => $fid,
+          ]
+        );
+      }
```

Comment:
Since most files are probably going to be clean (I hope, at least..) we probably don't have to log this. We may just end up filling up the logs.